docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,422 @@
+  - Key Generation
+  - OIDC URL Setup
+  - Set Service Account Signing Flags
+
+#### <a name='KeyGeneration'></a>Key Generation
+
+Admin should generate signing key pairs by using a tool such as openssl or bring their own public and private keys. 
+These keys will be mounted on a path on the containers running on the control plane node. These keys are required for signing the service account tokens that will be used by the capz pod. 
+
+#### <a name='OidcUrlSetup'></a>OIDC URL Setup
+
+Two documents i.e OIDC and JWKS json documents needs to be generated and published to a public URL. The OIDC document contains the metadata of the issuer. The JSON Web Key Sets (JWKS) document contains the public signing key(s) that allows AAD to verify the authenticity of the service account token.
+
+For more details on how to set up OIDC URL and more details on it, please g through this this [link](https://azure.github.io/azure-workload-identity/docs/installation/self-managed-clusters/oidc-issuer.html)
+
+The steps on a high level to setup is the following
+- Create an azure blob storage account.
+- Create a storage container.
+- Generate the OIDC and JWKS document.
+- The document should be accessible on the public accessible URL which will be used later.
+
+#### <a name='SetServiceAccountSigningFlags'></a>Set Service Account Signing Flags
+
+Setup the flags on the kind cluster. An example is shown below
+
+```yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  extraMounts:
+      # path on node where the public key exists
+    - hostPath: ${SERVICE_ACCOUNT_KEY_FILE}
+      containerPath: /etc/kubernetes/pki/sa.pub
+      # path on node where the private key exists
+    - hostPath: ${SERVICE_ACCOUNT_SIGNING_KEY_FILE}
+      containerPath: /etc/kubernetes/pki/sa.key
+  kubeadmConfigPatches:
+  - |
+    kind: ClusterConfiguration
+    apiServer:
+      extraArgs:
+        # the oidc url after it has been set up
+        service-account-issuer: ${SERVICE_ACCOUNT_ISSUER}
+        service-account-key-file: /etc/kubernetes/pki/sa.pub
+        service-account-signing-key-file: /etc/kubernetes/pki/sa.key
+    controllerManager:
+      extraArgs:
+        service-account-private-key-file: /etc/kubernetes/pki/sa.key
+```
+
+#### <a name='FederatedCredential'></a>Federated Credential
+
+The service account that the capz pod will use should be annotated with the client ID of the AAD application or User Assigned Identity and a federated identity should be created by associating the Service Account as a subject.
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    azure.workload.identity/client-id: ${APPLICATION_CLIENT_ID:-$USER_ASSIGNED_IDENTITY_CLIENT_ID}
+  labels:
+    azure.workload.identity/use: "true"
```

Comment:
This label in service account is no longer required starting from `v0.15.0` release. We've announced this as a breaking change in the `v1.0` alpha release: https://github.com/Azure/azure-workload-identity/releases/tag/v1.0.0-alpha.0. **tl;dr** This label is now required on the pod instead of the service account.

Reply:
This is nice. Thanks for the heads up. Will include this in the doc.